guides/search.md

```diff
@@ -0,0 +1,147 @@
+# Search
+OpenSearch provides a powerful search API that allows you to search for documents in an index. The search API supports a number of parameters that allow you to customize the search operation. In this guide, we will explore the search API and its parameters.
+
+## Setup
+Let's start by creating an index and adding some documents to it:
+
+```csharp
+    using OpenSearch.Client;
+    using OpenSearch.Net;
+    
+    var node = new Uri("https://localhost:9200");
+    var config = new ConnectionSettings(node)
+        .ThrowExceptions()
+        .ServerCertificateValidationCallback(CertificateValidations.AllowAll)
+        .BasicAuthentication("admin", "admin");
+    var client = new OpenSearchClient(config);
+    class Movie
+    {
+        public string Title { get; set; }
+        public string Director { get; set; }
+        public int Year { get; set; }
+        public override string ToString()
+        {
+            return $"{nameof(Title)}: {Title}, {nameof(Director)}: {Director}, {nameof(Year)}: {Year}";
+        }
+    }
+    await client.Indices.CreateAsync("movies", c => c
+        .Settings(s => s
+            .NumberOfShards(1)
+            .NumberOfReplicas(0)
+        )
+        .Map<Movie>(m => m
+            .Properties(p => p
+                .Text(t => t
+                    .Name(o => o.Title))
+                .Text(t => t
+                    .Name(o => o.Director))
+                .Number(n => n
+                    .Name(o => o.Year)
+                    .Type(NumberType.Integer)))));
+    var movies = new List<Movie>
+    {
+        new Movie { Title = "The Godfather", Director = "Francis Ford Coppola", Year = 1972 },
+        new Movie { Title = "The Shawshank Redemption", Director = "Frank Darabont", Year = 1994 },
+    };
+    for (var i = 0; i < 10; ++i)
+        movies.Add(new Movie { Title = "The Dark Knight " + i, Director = "Christopher Nolan", Year = 2008 + i });
+    // Index the movies
+    var indexResponse = await client.BulkAsync(b => b
+        .Index("movies")
+        .IndexMany(movies)
+        .Refresh(Refresh.WaitFor));
+    Console.WriteLine($"Indexed {indexResponse.Items.Count} movies"); // Indexed 12 movies
+
+    // Refresh the documents to make them searchable
+    var refreshResponse = await client.Indices.RefreshAsync("movies");
```

Comment:
This needs to be un-indented, and the final refresh call is no-longer needed when using a `Bulk` request with `Refresh.WaitFor`:
```suggestion
using OpenSearch.Client;
using OpenSearch.Net;

var node = new Uri("https://localhost:9200");
var config = new ConnectionSettings(node)
    .ThrowExceptions()
    .ServerCertificateValidationCallback(CertificateValidations.AllowAll)
    .BasicAuthentication("admin", "admin");
var client = new OpenSearchClient(config);

class Movie
{
    public string Title { get; set; }
    public string Director { get; set; }
    public int Year { get; set; }

    public override string ToString()
    {
        return $"{nameof(Title)}: {Title}, {nameof(Director)}: {Director}, {nameof(Year)}: {Year}";
    }
}

await client.Indices.CreateAsync("movies", c => c
    .Settings(s => s
        .NumberOfShards(1)
        .NumberOfReplicas(0)
    )
    .Map<Movie>(m => m
        .Properties(p => p
            .Text(t => t
                .Name(o => o.Title))
            .Text(t => t
                .Name(o => o.Director))
            .Number(n => n
                .Name(o => o.Year)
                .Type(NumberType.Integer)))));

var movies = new List<Movie>
{
    new Movie { Title = "The Godfather", Director = "Francis Ford Coppola", Year = 1972 },
    new Movie { Title = "The Shawshank Redemption", Director = "Frank Darabont", Year = 1994 },
};
for (var i = 0; i < 10; ++i)
    movies.Add(new Movie { Title = "The Dark Knight " + i, Director = "Christopher Nolan", Year = 2008 + i });

// Index the movies
var indexResponse = await client.BulkAsync(b => b
    .Index("movies")
    .IndexMany(movies)
    .Refresh(Refresh.WaitFor));
Console.WriteLine($"Indexed {indexResponse.Items.Count} movies"); // Indexed 12 movies
```